f5lbaasdriver/test/tempest/tests/api/test_l7policy_rules.py

```diff
@@ -129,19 +203,16 @@ def test_policy_reject_header_ends_with(self):
         self._create_l7rule(l7policy.get('id'), **rule_args)
 
         for bigip_client in self.bigip_clients:
-            assert bigip_client.policy_exists("wrapper_policy",
-                                              "Project_" +
-                                              self.project_tenant_id)
-            assert bigip_client.rule_exists("wrapper_policy",
-                                            "reject_1",
-                                            "Project_" +
-                                            self.project_tenant_id)
-            assert bigip_client.rule_has_condition("wrapper_policy",
-                                                   "reject_1",
-                                                   "endsWith",
-                                                   "real",
-                                                   "Project_" +
-                                                   self.project_tenant_id)
+            self.assertion_check(
```

Comment:
So in order to modify this test, I need to do:
self.wrapper_policy[0] += "_{}".format(listener_id)
self.reject_1[0] += "_{}".format(listener_id)
self.reject_1_endswith_real[0] += "_{}.format(listener_id)

Reply:
The named tuple will croak at doing what you're suggesting above... It sounds like we need to go for an object rather than using namedtuples for this solution.

That said, if you wanted an amendment to all arguments like this, then a change could easily be made to append a dynamic variable to each argument within besides the tenant_id for the project.